### PR TITLE
expression: add unit test for schema.go

### PR DESCRIPTION
**Pull request by zz-jason**
to https://github.com/pingcap/tidb/issues/3729

### CODE REVIEW

File: expression/schema_test.go [hunk 1]
```diff
@@ -0,0 +1,280 @@
+// Copyright 2015 PingCAP, Inc.
```
Comment (XuHuaiyu) [inline expression/schema_test.go]: s/2015/2017

Comment (zz-jason) [inline expression/schema_test.go]: done

**Comment by zz-jason** [comment]
@XuHuaiyu @hanfei1991 PTAL

**Comment by XuHuaiyu** [comment]
Can the duplicated code like
`initiation of schema`, `colOutSchema`
be extracted. @zz-jason

File: expression/schema_test.go [hunk 2]
```diff
@@ -20,16 +20,14 @@ import (
 	"github.com/pingcap/tidb/model"
 )
 
-func (s *testEvalSuite) TestSchemaString(c *C) {
-	colCount := 5
-	keyInfoCount := 3
+func GenerateSchema(colCount int, dbName, tblName, fromID string) *Schema {
```
Comment (XuHuaiyu) [inline expression/schema_test.go]: We only use it in this file,
may be this function can be unexported.

Same as line43. @zz-jason

**Pull request merged**